wagtail/embeds/tests/test_embeds.py

```diff
@@ -387,6 +425,126 @@ def test_endpoint_with_format_param(self, loads, urlopen):
         self.assertEqual(request.get_full_url().split('?')[0], "https://www.vimeo.com/api/oembed.json")
 
 
+class TestInstagramOEmbed(TestCase):
+    def setUp(self):
+        class DummyResponse:
+            def read(self):
+                return b"foo"
+        self.dummy_response = DummyResponse()
+
+    def test_instagram_oembed_only_accepts_new_url_patterns(self):
+        finder = InstagramOEmbedFinder()
+        self.assertTrue(finder.accept("https://www.instagram.com/p/CHeRxmnDSYe/?utm_source=ig_embed"))
+        self.assertFalse(finder.accept("https://instagr.am/p/CHeRxmnDSYe/?utm_source=ig_embed"))
+
+    @patch('urllib.request.urlopen')
+    @patch('json.loads')
+    def test_instagram_oembed_return_values(self, loads, urlopen):
+        urlopen.return_value = self.dummy_response
+        loads.return_value = {
```

Comment:
Patching json.loads in this way means that we end up not really testing any of the functionality of the finder - anything that calls `json.loads` (with any parameters) would pass the test. I'd suggest updating the definition of DummyResponse above so that it returns the expected JSON, and adding a `urlopen.assert_called_with` to confirm that we're using the right URL and credentials.

(Likewise with the Facebook test.)